Dont convert variables to numpy if on interface-specific device (#2136)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Bug fixes</h3>

> * Fixes a bug where you can't use `diff_method=None` with the jax device and jitting.

Reviewer: ```suggestion
* Fixes a bug where the `default.qubit.jax` device can't be used with `diff_method=None` and jitting.
```
